Firmware/communication/CircularBuffer.hpp

```diff
@@ -0,0 +1,333 @@
+/*******************************************************************************
+* File          : circularBuffer.hpp
+*
+* Description   : Implementation file for circular buffers
+*
+* Project       :
+*
+* Author        : S.Gilbert
+*
+* Created on    : 09 Dec 2019
+*
+*
+*******************************************************************************/
+/* Define to prevent recursive inclusion -------------------------------------*/
+#ifndef CIRCULAR_BUFFER_H
+#define CIRCULAR_BUFFER_H
+
+/*******************************************************************************
+INCLUDES
+*******************************************************************************/
+
+#include <string.h>
+#include <stddef.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+/*******************************************************************************
+NAMESPACE
+*******************************************************************************/
+
+/*******************************************************************************
+DEFINITIONS
+*******************************************************************************/
+
+typedef enum
+{
+	CB_OK = 0,
+	CB_FAIL,
+	CB_SIZE_ERROR,
+	CB_POINTER_ERROR
+}CB_Status_t;
```

Comment:
this seems to be unused

Reply:
This was a hold over from when the status of each function would be returned but I have removed this.